proc/community/src/test/java/org/neo4j/gds/leiden/LeidenWriteProcTest.java

```diff
@@ -233,4 +234,31 @@ void shouldEstimateMemory() {
                     "})";
         assertThatNoException().isThrownBy(() -> runQuery(query));
     }
+
+    static Stream<Arguments> communitySizeInputs() {
+        return Stream.of(
+                Arguments.of(1, new Long[]{3L, 6L}),
+                Arguments.of(10, new Long[]{})
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("communitySizeInputs")
+    void writeWithMinCommunitySize(int minCommunitySize, Long[] expectedCommunityIds) {
```

Comment:
We should also consider how this will interact with `includeIntermediateCommunities: true`.
Looking at the LeidenCompanion, I suspect the filter will not be applied right now.

My first thought would be, the the minCommunitySize could be only applied to the final community.
But also open for ideas here :)

Reply:
Thank you for your feedback :)

you are right, the filter is not being applied when `includeIntermediateCommunities: true`.

to apply the filter `consecutiveIds: true` must be set. having `includeIntermediateCommunities: true` set too brings the algorithm in an invalid state.

I think I completed the missing part of the issue; I will push and make this PR a non draft one.